### PR TITLE
Enhancement | Implement VPC Flow logs on EKS VPCs

### PR DESCRIPTION
**Pull request by Franr**
## What?
Adding VPC Flow logs to the EKS VPCs.
Controlled by a variable (`false` by default).

## Why?
For security and traceability reasons.

## References
https://github.com/binbashar/le-tf-infra-aws/issues/476

### CODE REVIEW

File: apps-devstg/us-east-1/k8s-eks-demoapps/network/network.tf [hunk 1]
```diff
@@ -131,3 +131,16 @@ resource "aws_route" "private_rt_routes_to_tgw" {
...
+module "vpc_flow_logs" {
+  source = "github.com/binbashar/terraform-aws-vpc-flowlogs.git?ref=v1.0.17"
+
+  # enable
+  count = var.enable_vpc_flow_logs ? 1 : 0
+
+  vpc_id             = module.vpc-eks.vpc_id
+  bucket_name_prefix = local.vpc_name
+  tags               = local.tags
+  enable_versioning  = true
```
Comment (Franr) [inline apps-devstg/us-east-1/k8s-eks-demoapps/network/network.tf:145]: if you feel it is worth it, I can add an extra variable to control the versioning of the bucket.

File: apps-devstg/us-east-1/k8s-eks/network/network.auto.tfvars [hunk 2]
```diff
@@ -1 +1,2 @@
 vpc_enable_nat_gateway = false
+enable_vpc_flow_logs = false
```
Comment (diego-ojeda-binbash) [inline apps-devstg/us-east-1/k8s-eks/network/network.auto.tfvars:2]: This looks redundant since the variable's default value is already set to false.

**Review by diego-ojeda-binbash — CHANGES_REQUESTED** [review]

File: apps-devstg/us-east-1/k8s-eks-demoapps/network/network.tf (hunk 1 above)
Comment (Franr) [reply apps-devstg/us-east-1/k8s-eks-demoapps/network/network.tf:145]: the value is "hardcoded", so you might want to make it dynamic with a variable (and just set values through `.auto.tfvars` files; of course you can always set it to `false` or whatever directly on the tf file, so it is just a cosmetic thing

Comment (diego-ojeda-binbash) [reply apps-devstg/us-east-1/k8s-eks-demoapps/network/network.tf:145]: Oh I see. I think for now it's okay as it is.

**Review by diego-ojeda-binbash — APPROVED** [review]

**Pull request merged**